extensions/smb/tests/ListSmbTests.cpp

```diff
@@ -0,0 +1,146 @@
+  auto hide_file_error = minifi::test::utils::hide_file(mock_smb_connection_controller_service->getPath() / "g.foo");
+  REQUIRE_FALSE(hide_file_error);
+
+  REQUIRE((a_expected_attributes && b_expected_attributes && c_expected_attributes && d_expected_attributes && e_expected_attributes && f_expected_attributes && g_expected_attributes));
+
+  REQUIRE(controller.plan->setProperty(list_smb, ListSmb::ConnectionControllerService, "smb_connection_controller_service"));
+
+  SECTION("FileFilter without subdirs") {
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::FileFilter, ".*\\.foo"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::RecurseSubdirectories, "false"));
+    const auto trigger_results = controller.trigger();
+    CHECK(trigger_results.at(ListSmb::Success).size() == 2);
+    CHECK_FALSE(list_smb->isYield());
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *a_expected_attributes));
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *b_expected_attributes));
+  }
+
+  SECTION("Input directory") {
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::InputDirectory, "subdir"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::RecurseSubdirectories, "true"));
+    const auto trigger_results = controller.trigger();
+    CHECK(trigger_results.at(ListSmb::Success).size() == 1);
+    CHECK_FALSE(list_smb->isYield());
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *d_expected_attributes));
+  }
+
+  SECTION("PathFilter and FileFilter") {
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::FileFilter, ".*\\.foo"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::RecurseSubdirectories, "true"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::PathFilter, "subdir.*"));
+    const auto trigger_results = controller.trigger();
+    CHECK(trigger_results.at(ListSmb::Success).size() == 2);
+    CHECK_FALSE(list_smb->isYield());
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *d_expected_attributes));
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *e_expected_attributes));
+  }
+
+  SECTION("Subdirs with age restriction") {
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::MinimumFileAge, "3min"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::MaximumFileAge, "59min"));
+    REQUIRE(controller.plan->setProperty(list_smb, ListSmb::RecurseSubdirectories, "true"));
+    const auto trigger_results = controller.trigger();
+    CHECK(trigger_results.at(ListSmb::Success).size() == 3);
+    CHECK_FALSE(list_smb->isYield());
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *a_expected_attributes));
+    CHECK(checkForFlowFileWithAttributes(trigger_results.at(ListSmb::Success), *d_expected_attributes));
+  }
```

Comment:
Is there a problem with this, or is the virtual machine the CI jobs run on simply too slow?

```
-------------------------------------------------------------------------------
ListSmb tests
  Subdirs with age restriction
-------------------------------------------------------------------------------
D:\a\nifi-minifi-cpp\nifi-minifi-cpp\extensions\smb\tests\ListSmbTests.cpp(107)
...............................................................................

D:\a\nifi-minifi-cpp\nifi-minifi-cpp\extensions\smb\tests\utils/MockSmbConnectionControllerService.h(50): FAILED:
  CHECK( std::chrono::abs(expected_creation_time - *creation_time_from_attribute) < 1s )
with expansion:
  10056573 [1/10000000] s < 1 s

D:\a\nifi-minifi-cpp\nifi-minifi-cpp\extensions\smb\tests\utils/MockSmbConnectionControllerService.h(51): FAILED:
  CHECK( std::chrono::abs(expected_last_access_time - *last_access_time_from_attribute) < 1s )
with expansion:
  10056573 [1/10000000] s < 1 s

===============================================================================
test cases:   2 |   1 passed | 1 failed
assertions: 165 | 163 passed | 2 failed
```

Reply:
From what I've gathered its was only due to the slow CI. I've increased the threshold in https://github.com/apache/nifi-minifi-cpp/pull/1634/commits/40dff807c58f68f2070c54613fb9db73cad3bb72